src/Test/Mendel/MutationOperator.hs

```diff
@@ -1,18 +1,113 @@
-{- |
-Module         : Test.Mendel.MutationOperator
-Description    : Definition of mutation operators
-
-This module provides various mutation operators which can be used
-to inject faults into Haskell modules.
--}
-module Test.Mendel.MutationOperator (MuOp (..)) where
-
-{- | A mutation operator which describes a semantic change that should be applied to
-a Haskell module.
--}
-data MuOp
-    = ReverseString
-    | ReverseClausesInPatternMatch
-    | SwapPlusMinus
-    | SwapIfElse
-    deriving (Show, Eq)
+{-#  LANGUAGE Rank2Types, FlexibleInstances, ConstraintKinds #-}
+-- | Mutation operators
+module Test.Mendel.MutationOperator (MuOp
+          , Mutable(..)
+          , (==>*)
+          , (*==>*)
+          , (~~>)
+          , mkMpMuOp
+          , same
+          , Module_
+          , Exp_
+          , Decl_
+         -- , GuardedRhs_
```

Comment:
If it doesn't work then you can delete all occurrences of `GuardedRhs`.

Reply:
I just don't know yet, how to do it without type error